jobs/async-upload/job/config.py

```diff
@@ -0,0 +1,314 @@
+from __future__ import annotations
+import base64
+import json
+import configargparse as cap
+from typing import Any, Dict, Mapping
+from pathlib import Path
+
+
+def _parser() -> cap.ArgumentParser:
+    """Parse command line arguments and config files"""
+    p = cap.ArgumentParser(
+        default_config_files=[],
+        auto_env_var_prefix="MODEL_SYNC_",
+        description="Synchronise AI models between OCI registries and/or S3 buckets",
+    )
+
+    # --- source ---
+    # s3
+    p.add("--source-type", choices=["s3", "oci"], default="s3")
+    p.add("--source-aws-bucket")
+    p.add("--source-aws-region")
+    p.add("--source-aws-access-key-id")
+    p.add("--source-aws-secret-access-key")
+    p.add("--source-aws-endpoint")
+    # OCI registry
+    p.add("--source-oci-uri")
+    p.add("--source-oci-username")
+    p.add("--source-oci-password")
+
+    # --- destination ---
+    # s3
+    p.add("--destination-type", choices=["s3", "oci"], default="s3")
+    p.add("--destination-aws-bucket")
+    p.add("--destination-aws-region")
+    p.add("--destination-aws-access-key-id")
+    p.add("--destination-aws-secret-access-key")
+    p.add("--destination-aws-endpoint")
+    # OCI registry
+    p.add("--destination-oci-uri")
+    p.add("--destination-oci-username")
+    p.add("--destination-oci-password")
+
+    # --- model-registry ---
+    p.add("--model-name")
+    p.add("--model-version")
```

Comment:
Can you kindly check https://github.com/kubeflow/model-registry/issues/1108#issuecomment-2880448765 is considered here?
Do you see alternatives or otherwise?

Reply:
Indeed, the model-related parameters here are not fully fleshed out, I suppose I should have dropped a TODO in there somewhere. 

This is currently a minimal implementation, mostly to get the initial config data-structure in place. I think as we work on the actual async upload part of this (the core business logic) then these parameters will be fleshed out a bit more.